client/lib/services/leaderboard.ts

```diff
@@ -0,0 +1,62 @@
+import { atom } from 'jotai';
+import { LeaderboardEntry } from '../types';
+import { toast } from '@/hooks/use-toast';
+import { useAtom } from 'jotai';
+import { ipAtom } from './api';
+import { useEffect } from 'react';
+import { useWebSocket } from './ws';
+
+const leaderboardAtom = atom<LeaderboardEntry[]>([]);
+
+export const useLeaderboard = () => {
+    const [ip] = useAtom(ipAtom);
+    const { ws } = useWebSocket();
+    const [leaderboard, setLeaderboard] = useAtom(leaderboardAtom);
+
+    const rankLeaderboard = (data: LeaderboardEntry[]) => {
+        data.sort((a, b) => b.score - a.score);
+    };
```

Comment:
rank leaderboard seems like a weird name for this as it sorts the leaderboard.  Consider renaming this to `sortLeaderboard`.

Also, the API for this function would be a bit nicer if you return the `data.sort(...)` (up to you if you wanna make this change)